### PR TITLE
Add tests and change StatusReporter.client to osconfig.Interface

### PR DESCRIPTION
**Pull request by michaelgugino**
Add status_test.go unit tests.  Also, change type of
StatusReporter.client to osconfig.Interface to allow for
utilizing fakeclient library.

### CODE REVIEW

**Comment by enxebre** [comment]
/hold
@michaelgugino can you have a separate commit for vendoring, thanks?

File: pkg/operator/status_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+					Type:   configv1.OperatorAvailable,
+					Status: osconfigv1.ConditionTrue,
+				},
+				{
+					Type:   configv1.OperatorFailing,
+					Status: osconfigv1.ConditionFalse,
+				},
+			},
+		},
+		{
```
Comment (ingvagabund) [inline pkg/operator/status_test.go]: This item is identical to the previous one. Is that intentional?

Comment (michaelgugino) [reply pkg/operator/status_test.go]: Good catch!  No, I don't know how that happend!

Comment (michaelgugino) [reply pkg/operator/status_test.go]: @ingvagabund fixed.

**Comment by michaelgugino** [comment]
@enxebre added separate commit for vendor.

**Pull request merged**